File: client/data_handler.py
from _cmd_.cmd_checker import cmd_check

HEADER = 64
FORMAT = 'utf-8'
class DataHandler:
    def __init__(self,socket):
        self.socket = socket

    def receive_data(self):
        data = self.socket.recv(HEADER).decode(FORMAT)
        is_cmd = False
        if data:
            data = data.strip('|')
            buffer_size = int(data)
            data = self.socket.recv(buffer_size).decode(FORMAT)
            is_cmd = cmd_check(data)
        if not is_cmd:
            return data

    def send_data(self, msg):
        msg = self.set_header(msg)
        self.socket.send(msg)

    def set_header(self,msg):
        val_len = len(str(len(msg)))
        length = str(len(msg)).encode(FORMAT)
        new_value = length + b' ' * (HEADER - val_len-1) + b'|' + msg.encode(FORMAT)
        return new_value

File: server/show_devices/data_handler.ts
import {cmd_check} from "./cmd_checkr"
import {client} from "./get_devices"

export class DataHandler{
    private HEADER : number
    private is_cmd : boolean
    private FORMAT : string
    constructor() {
        this.HEADER = 64
        this.is_cmd = false
        this.FORMAT = "UTF-8"
    }
    receive_data(data:Buffer){
        let raw_data = data.slice(this.HEADER).toString()
        const is_cmd = cmd_check(raw_data)
        if (!is_cmd) {
            console.log('main_func',raw_data)
        }
    }
    send_data(msg: string){
        const value = this.set_header(msg)
        client.write(value)
    }

    private set_header(msg:string){
        const msg_len = msg.length
        const length = msg_len.toString()
        let new_value = length + ' '.repeat(this.HEADER-length.length-1) + "|" + msg
        return new_value
    }
}